Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -17,35 +18,76 @@ def initialize(*)
       end
 
       def run
-        outdated_casks = casks(alternative: lambda {
+        self.class.upgrade_casks(
+          *casks,
+          force:          force?,
+          greedy:         greedy?,
+          dry_run:        dry_run?,
+          binaries:       binaries?,
+          quarantine:     quarantine?,
+          require_sha:    require_sha?,
+          skip_cask_deps: skip_cask_deps?,
+          verbose:        verbose?,
+        )
+      end
+
+      def self.upgrade_casks(
+        *casks,
+        force: nil,
+        greedy: nil,
+        dry_run: nil,
+        binaries: nil,
+        skip_cask_deps: nil,
+        verbose: nil,
+        quarantine: nil,
+        require_sha: nil
+      )
+        # TODO: Handle this in `CLI::Parser`.
+        binaries    = Homebrew::EnvConfig.cask_opts_binaries?    if binaries.nil?
+        dry_run     = Homebrew::EnvConfig.cask_opts_dry_run?     if dry_run.nil?
```

Comment:
This isn't supported in `HOMEBREW_CASK_OPTS`. It may currently work like that, but that is an implementation detail since we don't raise errors on invalid options.

Reply:
What options does `HOMEBREW_CASK_OPTS` support?

Reply:
In addition to those, `--[no]-binaries` (which should be handled here), and `--language` and all the `--*dir` flags (which are handled in `Cask::Config`).